src/common/src/com/microsoftopentechnologies/tooling/msservices/helpers/azure/rest/MobileServiceRestManager.java

```diff
@@ -0,0 +1,131 @@
+                            headers.get("x-ms-request-id").size() <= 0 ||
+                            headers.get("x-ms-request-id").get(0) == null) {
+                        throw new AzureCmdException(String.format("Status code %s: %s. Contains no valid request id",
+                                code,
+                                response.getMessage()),
+                                response.getContent());
+                    }
+
+                    String requestId = headers.get("x-ms-request-id").get(0);
+                    String operationPath = pollPath + requestId;
+                    executePollRequest(managementUrl, operationPath, sslConnectionProvider);
+                }
+
+                return response.getContent();
+            }
+        } catch (IOException e) {
+            throw new AzureCmdException(e.getMessage(), e);
+        }
+    }
+
+    private static void executePollRequest(@NotNull String managementUrl,
+                                           @NotNull String operationPath,
+                                           @NotNull HttpsURLConnectionProvider sslConnectionProvider)
+            throws IOException, AzureCmdException {
+        while (true) {
+            HttpsURLConnection sslConnection = sslConnectionProvider.getSSLConnection(managementUrl,
+                    operationPath, ContentType.Xml);
+            HttpResponse response = getResponse("GET", null, sslConnection);
+
+            int code = response.getCode();
+
+            if (code < 200 || code >= 300) {
+                throw new AzureCmdException(String.format("Error status code %s: %s", code, response.getMessage()),
+                        response.getContent());
+            } else {
+                String pollres = response.getContent();
+
+                try {
+                    NodeList nl = ((NodeList) XmlHelper.getXMLValue(pollres, "//Status", XPathConstants.NODESET));
+
+                    if (nl.getLength() > 0) {
+                        String status = nl.item(0).getTextContent();
+                        if (status.equals("Succeeded")) {
+                            break;
+                        } else if (!status.equals("InProgress")) {
+                            throw new AzureCmdException(
+                                    String.format("Invalid status reported while polling async request: %s", status),
+                                    pollres);
+                        }
+                    } else {
+                        throw new AzureCmdException("No status reported while polling async request.", pollres);
+                    }
+                } catch (Exception e) {
+                    throw new AzureCmdException(e.getMessage(), e);
+                }
+            }
+
+            // wait for a while otherwise Azure complains with a
+            // "too many requests received" error
+            try {
+                Thread.sleep(2000);
```

Comment:
This has been moved as it is from the previous implementation. We should review this behavior, but it might take some time to find the appropriate way to handle/detect the exception, and adjust the sleep value.